sql/core/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -218,3 +218,23 @@ case class StringEndsWith(attribute: String, value: String) extends Filter {
 case class StringContains(attribute: String, value: String) extends Filter {
   override def references: Array[String] = Array(attribute)
 }
+
+/**
+ * A filter that always evaluates to `true`.
+ */
+case class AlwaysTrue() extends Filter {
```

Comment:
why it's a case class instead of object?

Reply:
There's an object that extends it just below. I thought it was still a good idea to match the pattern of the other filters and make a class as well as an object, since this will be used by Java code. Checking `instanceof AlwaysTrue$` leaks Scala details into Java.